src/provider-upgrade.ts

```diff
@@ -3,6 +3,7 @@ import { JobPermission } from "projen/lib/github/workflows-model";
 
 interface ProviderUpgradeOptions {
   checkForUpgradesScriptPath: string;
+  workflowRunsOn?: string[];
```

Comment:
Nit: making this required ensures we never forget to pass it down when we refactor this in the future

Reply:
yeah, nice idea. Done ✅